gooddata-sdk/gooddata_sdk/catalog/data_source/service.py

```diff
@@ -80,3 +86,59 @@ def register_upload_notification(self, data_source_id: str) -> None:
 
     def get_declarative_data_sources(self) -> CatalogDeclarativeDataSources:
         return CatalogDeclarativeDataSources.from_api(self._layout_api.get_data_sources_layout())
+
+    def put_declarative_data_sources(
+        self,
+        declarative_data_sources: CatalogDeclarativeDataSources,
+        credentials_path: Optional[Path] = None,
+        test_flag: bool = False,
```

Comment:
I would rename test_flag to something more descriptive - maybe test_data_sources?

Reply:
Agree!